tests/network/vmi_slirp_interface.go

```diff
@@ -170,12 +171,14 @@ var _ = SIGDescribe("Slirp Networking", func() {
 				dns = flags.ConnectivityCheckDNS
 			}
 
-			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
-				&expect.BSnd{S: "\n"},
-				&expect.BExp{R: console.PromptExpression},
-				&expect.BSnd{S: fmt.Sprintf("curl -o /dev/null -s -w \"%%{http_code}\\n\" -k https://%s\n", dns)},
-				&expect.BExp{R: "301"},
-			}, 180)).To(Succeed())
+			Eventually(func() error {
+				return console.SafeExpectBatch(vmi, []expect.Batcher{
+					&expect.BSnd{S: "\n"},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: fmt.Sprintf("curl -o /dev/null -s -w \"%%{http_code}\\n\" -k https://%s\n", dns)},
+					&expect.BExp{R: "301"},
+				}, 180)
+			}, 180*time.Second, time.Second).Should(Succeed(), "Failed to establish a successful connection to the outside network")
```

Comment:
I'm not sure if an interval of 1sec is enough to get a response, and it may also be too much load (in this case, we will try 3min, ~180 times to issue the command).
I am not even sure what happens if the body of the eventually is returning only after 5sec. The interval will just not operate during this period?

I would also change the batch timeout to something more reasonable, making sure the interval is higher than that.

Reply:
I don't know what to do about the first part of your comment. IIUIC the poll interval will start counting down after the body returns. It is not the maximum time for the body to complete. Any suggestion?

I changed the batch timeout to 60 seconds.

Reply:
>  IIUIC the poll interval will start counting down after the body returns.

I have not understood it from the text.
What made you understand it like this?

Reply:
https://github.com/onsi/gomega/blob/7b8b801e5f98d6c1168aff01100a59f6070d2b3e/internal/gomega.go#L56
https://github.com/onsi/gomega/blob/7b8b801e5f98d6c1168aff01100a59f6070d2b3e/internal/async_assertion.go#L292

Reply:
If I read the logic correctly, in the case we get as a response to the `curl` command a different code/error, then we will block for 60sec (continuing to wait for "301"). Therefore, in practice only 3 attempts will occur.
Is that what you intended?

Reply:
Oh, now I see. I did not think of that. I would keep it on one minute. The curl command is failing very rarely. For those rare cases, I'd wait one minute. I'm worried that if we make the batch timeout to small, it would timeout on slow systems with terrible connectivity. Do you think that's ok?

Reply:
I think it can be improved, but this is an improvement already and we should take it.
Thanks!